File: examples/ts-scripts/pretty_streaming.ts
/**
 * Example script demonstrating streaming requests to an Ollama server.
 * Displays a chatbot-style interface with real-time streaming responses.
 */

import { Ollama } from "ollama";

function printHeader(): void {
  console.log("\n" + "=".repeat(60));
  console.log("  🤖 Ollama Streaming Chat");
  console.log("=".repeat(60) + "\n");
}

function printUserMessage(prompt: string, model: string): void {
  console.log(`📝 You: ${prompt}`);
  console.log(`🧠 Model: ${model}\n`);
  process.stdout.write("💬 Assistant: ");
}

async function streamResponse(
  client: Ollama,
  model: string,
  prompt: string
): Promise<void> {
  const response = await client.generate({
    model: model,
    prompt: prompt,
    stream: true,
  });

  for await (const chunk of response) {
    process.stdout.write(chunk.response);
  }
}

function printFooter(): void {
  console.log("\n\n" + "-".repeat(60));
  console.log("✓ Streaming response completed.");
  console.log("-".repeat(60) + "\n");
}

async function main(): Promise<void> {
  // Initialize the Ollama client
  const client = new Ollama({ host: "http://127.0.0.1:5001" });

  // Model and prompt to use
  const model = "mistral";
  const prompt = "Why is the sky blue?";

  // Display the interface
  printHeader();
  printUserMessage(prompt, model);

  // Stream the response
  await streamResponse(client, model, prompt);

  printFooter();
}

main().catch(console.error);

File: examples/pretty_streaming.py
#!/usr/bin/env python3
"""
This script will send requests to the Oshepherd API, which will orchestrate Ollama models inference execution
on available Oshepherd Workers using streaming mode.
"""

import ollama


def print_header():
    """Display the application header."""
    print("\n" + "=" * 60)
    print("  🤖 Ollama Streaming Chat")
    print("=" * 60 + "\n")


def print_user_message(prompt, model):
    """Display the user's message and model info."""
    print(f"📝 You: {prompt}")
    print(f"🧠 Model: {model}\n")
    print("💬 Assistant: ", end="", flush=True)


def stream_response(client, model, prompt):
    """Stream and display the response in real-time."""
    for chunk in client.generate(model=model, prompt=prompt, stream=True):
        response_text = chunk['response']
        print(response_text, end='', flush=True)


def print_footer():
    """Display the completion footer."""
    print("\n\n" + "-" * 60)
    print("✓ Streaming response completed.")
    print("-" * 60 + "\n")


def main():
    # Initialize the Ollama client
    client = ollama.Client(host="http://127.0.0.1:5001")

    # Model and prompt to use
    model = "mistral"
    prompt = "Why is the sky blue?"

    # Display the interface
    print_header()
    print_user_message(prompt, model)

    # Stream the response
    stream_response(client, model, prompt)

    print_footer()


if __name__ == "__main__":
    main()
